### DISCUSSION TITLE
github actions manager is incorrectly downgrading an action (digest)

### DISCUSSION THREAD

**Discussion by noahnu** [renovatebot/renovate#22739] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I am trying to use renovate to keep GitHub Actions up to date using the digest update strategy. I have a github workflow using [hashicorp/setup-tlfint](https://github.com/terraform-linters/setup-tflint) @ `eaeaad19ca755b477f6a5b99c776612b73ee6b3f`. Renovate keeps on putting up a PR to "update" from `eaeaad19ca755b477f6a5b99c776612b73ee6b3f` to https://github.com/terraform-linters/setup-tflint/commit/db27e99. That's an older version.

It looks like it's somehow picking a digest from a tag, even though this config works for other github actions just fine. Not sure what's special about setup-tflint.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: detectSemanticCommits()
DEBUG: getCommitMessages
DEBUG: semanticCommits: detected "unknown"
DEBUG: semanticCommits: disabled
DEBUG: 1 flattened updates found: hashicorp/setup-tflint
DEBUG: Returning 1 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "baseBranch": "main"
  "config": {
    "github-actions": [
      {
        "deps": [
          {
            "depName": "actions/checkout",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "replaceString": "actions/checkout@96f53100ba2a5449eb71d2e6604bbcd94b9449b5",
            "autoReplaceStringTemplate": "{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # {{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}",
            "currentDigest": "96f53100ba2a5449eb71d2e6604bbcd94b9449b5",
            "updates": [],
            "packageName": "actions/checkout",
            "warnings": []
          },
          {
            "depName": "hashicorp/setup-tflint",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "replaceString": "hashicorp/setup-tflint@eaeaad19ca755b477f6a5b99c776612b73ee6b3f",
            "autoReplaceStringTemplate": "{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # {{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}",
            "currentDigest": "eaeaad19ca755b477f6a5b99c776612b73ee6b3f",
            "updates": [
              {
                "updateType": "digest",
                "newDigest": "db27e99508f896b3c2a26580bbe3ef7df97d48c7",
                "branchName": "renovate/github-actions"
              }
            ],
            "packageName": "hashicorp/setup-tflint",
            "warnings": []
          }
        ],
        "packageFile": ".github/workflows/workflow.yml"
      }
    ]
  }
}
```

</details>

- Minimal reproduction: https://github.com/noahnu/renovate-github-actions-bug/pull/1
- Logs: https://developer.mend.io/github/noahnu/renovate-github-actions-bug/-/job/0188b4e4-1e95-751d-9698-2aa6e4097582

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
It looks to be the latest commit in the repo:

<img width="1029" alt="image" src="https://github.com/renovatebot/renovate/assets/6311784/cc8a6fcd-9db7-4234-8259-98af39350655">

The links you provided go to the repo it was forked from

**Reply by noahnu** [reply]
Oh wow. Not sure how I missed that. 😅
